Src/AmrCore/AMReX_Interpolater.cpp

```diff
@@ -729,84 +728,46 @@ CellConservativeQuartic::CoarseBox (const Box&     fine,
                                  FArrayBox&        fine,
-                                 int               fine_comp,
+                                 int               /* fine_comp */,
                                  int               ncomp,
                                  const Box&        fine_region,
                                  const IntVect&    ratio,
                                  const Geometry&   /* crse_geom */,
                                  const Geometry&   /* fine_geom */,
-                                 Vector<BCRec> const&   bcr,
-                                 int               actual_comp,
-                                 int               actual_state,
-                                 RunOn             /*runon*/)
+                                 Vector<BCRec> const& bcr,
+                                 int               /* actual_comp */,
+                                 int               /* actual_state */,
+                                 RunOn             runon)
 {
     BL_PROFILE("CellConservativeQuartic::interp()");
     BL_ASSERT(bcr.size() >= ncomp);
-    BL_ASSERT(ratio[0]==2);
+    BL_ASSERT(ratio[0] == 2);
 #if (AMREX_SPACEDIM >= 2)
     BL_ASSERT(ratio[0] == ratio[1]);
 #endif
 #if (AMREX_SPACEDIM == 3)
     BL_ASSERT(ratio[1] == ratio[2]);
 #endif
+    amrex::ignore_unused(bcr,ratio);
 
     //
     // Make box which is intersection of fine_region and domain of fine.
     //
     Box target_fine_region = fine_region & fine.box();
-    //
-    // crse_bx is coarsening of target_fine_region, grown by 2.
-    //
-    Box crse_bx = CoarseBox(target_fine_region,ratio);
-
-    Box crse_bx2(crse_bx);
-    crse_bx2.grow(-2);
-    Box fine_bx2 = amrex::refine(crse_bx2,ratio);
-
-    Real* fdat       = fine.dataPtr(fine_comp);
-    const Real* cdat = crse.dataPtr(crse_comp);
-
-    const int* flo    = fine.loVect();
-    const int* fhi    = fine.hiVect();
-    const int* clo    = crse.loVect();
-    const int* chi    = crse.hiVect();
-    const int* fblo   = target_fine_region.loVect();
-    const int* fbhi   = target_fine_region.hiVect();
-    const int* cblo   = crse_bx.loVect();
-    const int* cbhi   = crse_bx.hiVect();
-    const int* cb2lo  = crse_bx2.loVect();
-    const int* cb2hi  = crse_bx2.hiVect();
-    const int* fb2lo  = fine_bx2.loVect();
-    const int* fb2hi  = fine_bx2.hiVect();
 
-    Vector<int> bc     = GetBCArray(bcr);
-    const int* ratioV = ratio.getVect();
-
-    int ltmp = fb2hi[0]-fb2lo[0]+1;
-    Vector<Real> ftmp(ltmp);
-
-#if (AMREX_SPACEDIM >= 2)
-    ltmp = (cbhi[0]-cblo[0]+1)*ratio[1];
-    Vector<Real> ctmp(ltmp);
-#endif
-
-#if (AMREX_SPACEDIM == 3)
-    ltmp = (cbhi[0]-cblo[0]+1)*(cbhi[1]-cblo[1]+1)*ratio[2];
-    Vector<Real> ctmp2(ltmp);
-#endif
+    // Extract pointers to fab data
+    Array4<Real const> const& crsearr = crse.const_array();
+    Array4<Real>       const& finearr = fine.array();
```

Comment:
```suggestion
    Array4<Real const> const& crsearr = crse.const_array(crse_comp);
    Array4<Real>       const& finearr = fine.array(fine_comp);
```
This is where `crse_comp` and `fine_comp` are needed.